ui/ui-components/components/loadingButton.tsx

```diff
@@ -1,13 +1,23 @@
-import * as React from "react";
-import { Button } from "react-bootstrap";
+import { Button, ButtonProps } from "react-bootstrap";
+import { GrouparooUIEdition, grouparooUiEdition } from "../utils/uiEdition";
 import Loader from "./loader";
 
-export default function LoadingButton(props) {
-  const message = props.disabled ? (
-    <Loader size="sm" />
-  ) : (
-    props.children || "Submit"
-  );
+export default function LoadingButton(
+  props: ButtonProps & {
+    displayOn?: Array<GrouparooUIEdition>;
+    hideOn?: Array<GrouparooUIEdition>;
```

Comment:
```suggestion
    displayOn?: Array<GrouparooUIEdition>;
    hideOn?: Array<GrouparooUIEdition>;
```
I don't know if this is no longer the case, but I seem to recall Typescript prefering `Type[]` over `Array<Type>`